### ISSUE TITLE
spack uninstall leaves behind created module/dotkit

### ISSUE THREAD

**Issue by malcook** [LLNL/spack#44]
One might reasonable expect uninstall to completely undo the results of install.

But, I not that uninstall does not touch and installed modules.

For instance:

```shell
spack install dos2unix@5.3
   ...
==> Successfully installed dos2unix.
   ...
$ module av 2>&1 | grep dos2unix
dos2unix@5.3%gcc@4.8.2
$ which dos2unix
/usr/bin/dos2unix
$ module load dos2unix@5.3%gcc@4.8.2
$ which dos2unix
/n/local/sce/spack/opt/CentOS_7_x86_64/gcc@4.8.2/dos2unix@5.3/bin/dos2unix
$ module unload dos2unix@5.3%gcc@4.8.2
$ which dos2unix
/usr/bin/dos2unix
$ spack uninstall dos2unix
==> Successfully uninstalled dos2unix@5.3%gcc@4.8.2=CentOS_7_x86_64.
$ module av 2>&1 | grep dos2unix
dos2unix@5.3%gcc@4.8.2
$ module load dos2unix@5.3%gcc@4.8.2
$ which dos2unix
/usr/bin/dos2unix
```

I am not sure what the best resolution of this is.  Feel free to close with "works as designed".  I thought I should mention it though...

**Comment by adamjstewart** [comment]
Is this fixed by @alalazo's #754?

**Comment by tgamblin** [comment]
It is.  Thanks!
